### PR TITLE
Update streaming section with parallel promises

### LINKED ISSUE #3333: Update streaming docs to reflect Parallelized rendering
## What?
With Astro 2.5 [Parallelized rendering](https://astro.build/blog/astro-250/#parallelized-rendering), it's worth adding some more info here: 

CleanShot 2023-05-25 at 13 21 11@2x

CleanShot 2023-05-25 at 13 21 55@2x

See https://docs.astro.build/en/guides/server-side-rendering/#streaming

## Why? 
Data-loading components higher in the tree (siblings) will no longer block other components also loading data ...

### PR DESCRIPTION
**Pull request by sasoria**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes #3333

This Pull request updates the promise section related to streaming with parallel rendring. I ran the promises in dev mode to see how they would render in a template.

### CODE REVIEW

File: src/content/docs/en/guides/server-side-rendering.mdx [hunk 1]
```diff
@@ -365,4 +365,5 @@ const factPromise = fetch('https://catfact.ninja/fact')
 </html>
 ```
 
-In this example, `A name` will render while `personPromise` is loading. At that point, `A fact` will be appear, while `factPromise` is loading.
+In this example, `A name` will render while `personPromise` and `factPromise` are loading.
+Once `personPromise` has resolved `A fact` will be appear and `factPromise` will render when its finished loading.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/server-side-rendering.mdx:369]: I did a small wording change and fixed a typo, let me know what you think!
```suggestion
Once `personPromise` has resolved, `A fact` will appear and `factPromise` will render when it's finished loading.
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks @sasoria! Added a suggestion for you 🙌

File: src/content/docs/en/guides/server-side-rendering.mdx (hunk 1 above)
Comment (sasoria) [reply src/content/docs/en/guides/server-side-rendering.mdx]: Love it, I've added a commit the suggestions. Thanks for the feedback.

**Review by ematipico — APPROVED** [review]

**Comment by sarah11918** [comment]
Just updated branch because the Netlify deploys failed, and maybe this will trigger them again. (Otherwise, I can do it manually. Just doing this first)

**Review by sarah11918 — APPROVED** [review]
Approving for FriYAY merge day!

**Review by Yan-Thomas — APPROVED** [review]

**Pull request merged**